mautic/mautic
# Use Guzzle client instead of another package

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-470



#### Description:

The goal of this PR is to remove this deprecated package that is highlighted on `composer install`:
```
Package php-http/message-factory is abandoned, you should avoid using it. Use psr/http-factory instead.
```
It's needed for a Sparkpost test so the deprecated package can be removed together with `php-http/mock-client` dev dependency once the Sparkpost transport is moved from core to its own repository once https://github.com/mautic/mautic/pull/12423 is merged.

It was simple to replace this client with Guzzle's client in other tests.

Todo:
- [x] Merge https://github.com/mautic/mautic/pull/12423
- [x] Remove `php-http/mock-client` dev dependency

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. No need to test as it's used only in the tests. No production feature was touched.
